pkg/ebpf/tracer.go

```diff
@@ -196,10 +199,44 @@ func NewFlowFetcher(cfg *FlowFetcherConfig) (*FlowFetcher, error) {
 		pktDropsTracePoint:       pktDropsLink,
 		rttFentryLink:            rttFentryLink,
 		rttKprobeLink:            rttKprobeLink,
+		egressTCXLink:            map[ifaces.Interface]link.Link{},
+		ingressTCXLink:           map[ifaces.Interface]link.Link{},
 		lookupAndDeleteSupported: true, // this will be turned off later if found to be not supported
 	}, nil
 }
 
+func (m *FlowFetcher) AttachTCX(iface ifaces.Interface) error {
+	ilog := log.WithField("iface", iface)
+	if iface.NetNS != netns.None() {
+		if err := unix.Setns(int(iface.NetNS), unix.CLONE_NEWNET); err != nil {
+			return fmt.Errorf("failed to setns to %s: %w", iface.NetNS, err)
+		}
+	}
+	egrLink, err := link.AttachTCX(link.TCXOptions{
```

Comment:
I guess this function should honor the `FlowFetcher` `enableIngress` and `enableEgress` flags, like `Register` does?

Also that makes me think, as the agent can already be configured to capture only ingress or only egress via `config.Direction`, the filter `config.FlowFilterDirection` recently added was probably not necessary? Can we remove it? (not asking that for this PR)

Reply:
> Also that makes me think, as the agent can already be configured to capture only ingress or only egress via config.Direction, the filter config.FlowFilterDirection recently added was probably not necessary? Can we remove it? (not asking that for this PR)

Ok forget about it .. as discussed in slack, these two settings are redundant in ACCEPT filter mode, but they are not in REJECT mode. E.g. we may capture all ingress+egress globally but filter out a small subset of flows that are ingress or egress, for example on a particular port or subnet.